code/game/objects/effects/temporary_visuals/miscellaneous.dm

```diff
@@ -543,3 +543,36 @@
 	icon_state = "light_ash"
 	icon = 'icons/effects/weather_effects.dmi'
 	duration = 3.2 SECONDS
+
+/obj/effect/temp_visual/sonar_ping
+	duration = 3 SECONDS
+	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	anchored = TRUE
+	randomdir = FALSE
+	/// The image shown to modsuit users
+	var/image/modsuit_image
+	/// The person in the modsuit at the moment, really just used to remove this from their screen
+	var/datum/weakref/mod_man
+	/// The icon state applied to the image created for this ping.
+	var/real_icon_state = "sonar_ping"
+
+/obj/effect/temp_visual/sonar_ping/Initialize(mapload, mob/living/looker, mob/living/creature)
+	. = ..()
+	modsuit_image = image(icon = icon, loc = looker.loc, icon_state = real_icon_state, layer = ABOVE_ALL_MOB_LAYER, pixel_x = ((creature.x - looker.x) * 32), pixel_y = ((creature.y - looker.y) * 32))
+	modsuit_image.plane = ABOVE_LIGHTING_PLANE
+	mod_man = WEAKREF(looker)
+	add_mind(looker)
+
+/obj/effect/temp_visual/sonar_ping/Destroy()
+	remove_mind(mod_man.resolve())
```

Comment:
what if mod_man doesn't exist?

Reply:
i suggest you do something like
var/mob/living/previous_user = mod_man.resolve()
if(previous_user)
remove_mind(previous_user)

Reply:
Sure thing, I'll do it when I get home